### ISSUE TITLE
SegGptImageProcessor broken for 2D prompt masks that are not PIL images

### ISSUE THREAD

**Issue by EduardoPach** [huggingface/transformers#29549]
### System Info

- `transformers` version: 4.39.0.dev0
- Platform: macOS-14.2.1-arm64-arm-64bit
- Python version: 3.11.7
- Huggingface_hub version: 0.20.2
- Safetensors version: 0.4.1
- Accelerate version: 0.25.0
- Accelerate config: 	not found
- PyTorch version (GPU?): 2.1.2 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

The bug happens at this [line](https://github.com/huggingface/transformers/blob/main/src/transformers/models/seggpt/image_processing_seggpt.py#L351) because we expect 3-dimensional input, however, prompt masks can be 2D (semantic map) or 3D (in case a palette was already applied). This, however, is just a problem when passing `prompt_masks` as something different than a `PIL.Image`

```python
from transformers import AutoImageProcessor
import numpy as np
model_id = "BAAI/seggpt-vit-large"
image_processor = AutoImageProcessor.from_pretrained(model_id)
img_2d = np.random.rand(18,18)
_ = image_processor(images=None, prompt_masks=img_2d, return_tensors="pt")
```

### Expected behavior

It should work for every valid image input type rather than just for `PIL.Image`
